### PR TITLE
Add contribution guide

### PR DESCRIPTION
**Pull request by bkircher**
Base is from https://github.com/anselmh/CONTRIBUTING.md with some
adjustments for ews-cpp:
- add clang-format remarks
- add git pull --rebase remarks
- remove stuff not necessary for us

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+We like to encourage you to contribute to the repository.  This should be as
+easy as possible for you but there are a few things to consider when
+contributing.  The following guidelines for contribution should be followed if
+you want to submit a pull request.
+
+## How to prepare
+
+* You need a [GitHub account](https://github.com/signup/free)
+* Submit an [issue ticket](https://github.com/otris/ews-cpp/issues) for your
+  issue if there is no one yet.
```
Comment (m-Schlitzer) [inline CONTRIBUTING.md]: *if there isn't one yet.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Make Changes
+
+* In your forked repository, create a topic branch for your upcoming patch.
+  (e.g. `feature-autodiscover` or `bugfix-windows-crash`)
+  * Usually this is based on the master branch.
+  * Create a branch based on master; `git branch
+  my-work master` then checkout the new branch with `git
+  checkout my-work`.  Please avoid working directly on the `master` branch.
+* Make sure you stick to the coding style that is used already.
```
Comment (m-Schlitzer) [inline CONTRIBUTING.md]: *that is already in use.

**Comment by bkircher** [comment]
Thanks for catching these. Done

**Comment by m-Schlitzer** [comment]
Well then let's merge it.

**Pull request merged**